### ISSUE TITLE
AWS Batch S3 upload function does not work if output is a directory

### ISSUE THREAD

**Issue by MaxUlysse** [nextflow-io/nextflow#539]
Expected outputs:
```
output:
    set file("*multiqc_report.html"), file("*multiqc_data") into multiQCReport
```
Error:
```
Caused by:
  Missing output file(s) `*multiqc_data` expected by process `RunMultiQC`
```
There is a `multiqc_report.html` but it seems that the `multiqc_data` directory is not found.

here are the logs:
[nextflow.command.log](https://github.com/nextflow-io/nextflow/files/1517975/nextflow.command.log)
[task.command.log](https://github.com/nextflow-io/nextflow/files/1517979/task.command.log)
[task.command.err](https://gist.github.com/MaxUlysse/d2ee1978e1bef760d21f8f0790775a57)
[task.command.out](https://gist.github.com/MaxUlysse/53700cfacf3b7a0009d00598d84c7149)
[task.command.run](https://gist.github.com/MaxUlysse/7df7424d33e02604b9ba7aade6f1a239)

**Comment by pditommaso** [comment]
In the task wrapper there this command at the end 

```
nxf_s3_upload '*multiqc_data' s3://caw-test-results/work/ce/562f50f7277af34c4dc7d39e727612
```

Does that folder exist in the S3 path `s3://caw-test-results/work/ce/562f50f7277af34c4dc7d39e727612` ?

**Comment by MaxUlysse** [comment]
Not at all

**Comment by pditommaso** [comment]
so there's a problem in the upload, I will check it.

**Comment by MaxUlysse** [comment]
Thanks again.

**Comment by pditommaso** [comment]
how is the name of that directory ?

**Comment by MaxUlysse** [comment]
the directory is just `multiqc_data`, I copied the output directive from NGI-RNAseq, and they include possibilities of personnalization, so at some point it could be `anything_multiqc_data` I guess.

**Comment by pditommaso** [comment]
Shit! is a side effect of #535

**Comment by MaxUlysse** [comment]
Sorry about that

**Comment by pditommaso** [comment]
that's life! :)

**Comment by pditommaso** [comment]
But we have already a patch for it! Please try version `0.26.3-SNAPSHOT`

**Comment by MaxUlysse** [comment]
Still the same problem.

**Comment by pditommaso** [comment]
Can you include the command.run?

**Comment by MaxUlysse** [comment]
https://gist.github.com/MaxUlysse/fd038806878bb29835809863a578a7bb

**Comment by pditommaso** [comment]
The wrapper does not contain the patch https://github.com/nextflow-io/nextflow/commit/613b30a7251cb36e70581d533dcbe5a28ef56dad. 

It should be `eval "ls -d $pattern"` instead of `eval "ls $pattern"` 

Make sure you are using 

```
$ nextflow info
  Version: 0.26.3-SNAPSHOT build 4739
  Modified: 30-11-2017 14:23 UTC (15:23 CEST)
``

**Comment by MaxUlysse** [comment]
I'm using `NXF_VER=0.26.3-SNAPSHOT` already. How can I specify to use a more recent version?

**Comment by pditommaso** [comment]
does the build number match with mine?

**Comment by MaxUlysse** [comment]
not at all:
```
NXF_VER=0.26.3-SNAPSHOT nextflow info
  Version: 0.26.3-SNAPSHOT build 4738
  Modified: 28-11-2017 15:16 UTC (16:16 CEST)
...
```

**Comment by pditommaso** [comment]
if so do the following 

```
CAPSULE_RESET=1 NXF_VER=0.26.3-SNAPSHOT nextflow info
```

It will force to update your local snapshot. 

Then remove `CAPSULE_RESET=1 ` for the following runs

**Comment by MaxUlysse** [comment]
Thanks a lot, I got the same build now.
Trying at the moment, will tell you how it goes in a moment

**Comment by pditommaso** [comment]
:crossed_fingers:

**Comment by MaxUlysse** [comment]
It worked without any problem.
Thanks a lot for everything.

**Comment by pditommaso** [comment]
thanks to you for the troubleshooting! 🎉 🎉 🎉

**Comment by pditommaso** [comment]
Included in version `0.26.3`.
